### ISSUE TITLE
nextTick: Add note about long operations blocking UI refresh ins some browsers.

### ISSUE THREAD

**Issue by LinusBorg** [vuejs/vuejs.org#814]
As discussed [here](https://github.com/vuejs/vue/issues/5136#issuecomment-285288733), there's a caveat with $nextTick (or rather, native Promises as a whole, which $nextTick relies on):

The callback is pushed into the browsers microtask queue rather then the event loops normal queue. In some browsers (i.e. Chrome), than means that UI updates (due to DOM Changes) will be blocked until the microtask queue is done. 

That means that heavy operations in a `$nextTick` callback can still block the UI - while an "old-school" `setTimeout(cb, 0)` will have UI updates run before the `cb` is picked up by the event loop.

So in short:  

* use $nextTick when you have to do some lightweight (or really async, like AJAX) operation after the DOM has been updated.
* don't use `$nextTick` or rather, combine it with a `setTimeout(cb, 0)`, when you have heavy, sync operations to do.

**Comment by chrisvfritz** [comment]
Closing, since this is no longer an issue in 2.4+.
